benchmark/e2e/Qwen3.5/run_blocks.py

```
@@ -0,0 +1,182 @@
+    for label, block_cls in (("linear_attn (GDN+conv+MLP)", LinearAttentionBlock), ("full_attn (SDPA+MLP)", FullAttentionBlock)):
+        block = block_cls(cfg).to(dev).to(torch.bfloat16).train()
+        fwd, fwd_bwd = time_block(block, residual_stream, iters=args.iters)
+        measured[label] = (fwd, fwd_bwd)
+        del block
+        torch.cuda.empty_cache()
+
+    head = Head(cfg).to(dev).to(torch.bfloat16).train()
+    ids = torch.randint(0, cfg["vocab_size"], (args.bs, args.seq), device=dev)
+
+    def head_step():
+        head.zero_grad(set_to_none=True)
```

Ruff (0.16.2)

[error] Undefined name `head`

(F821)